app/src/main/java/samyups/example/trainingss/data/MainRepository.kt

```diff
@@ -0,0 +1,31 @@
+package samyups.example.trainingss.data
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+
+class MainRepository private constructor () {
+
+    private val dogsList = mutableListOf<Dog>()
+    private val dogs = MutableLiveData<List<Dog>>()
+
+    init {
+        dogs.value = dogsList
+    }
+
+    fun getDogsList() = dogs as LiveData<List<Dog>>
```

Comment:
Hm, I think we shouldn't cast this to `LiveData` in the `Repository`. My thinking is that the `Repository` doesn't know that we are using a `ViewModel` to do lifecycle-aware updates. Its job is to add data to the storage (currently only in memory) and read data from storage and pass it along to whoever wants it (in our case the `ViewModel`). We should be supplying a raw `List<Dog>` from this method, and then the `ViewModel` should worry about casting it to `LiveData<List<Dog>>`. Thoughts?